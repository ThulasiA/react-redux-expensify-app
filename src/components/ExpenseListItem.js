import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({id, description, amount, createdAt}) => (
  <div>
    <h4><Link to={`/edit/${id}`}>{description}</Link></h4>
    <p>{amount} - {createdAt}</p>
  </div>
);

export default ExpenseListItem;