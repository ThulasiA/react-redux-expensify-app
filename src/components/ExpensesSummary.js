import React from 'react';
import { connect } from 'react-redux';
import selectExpense from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
  const expensesWord = expensesCount > 1 ? 'expenses' : 'expense';
  const finalExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
  return (    
    <div>
      <p>Viewing {expensesCount} {expensesWord} totalling {finalExpensesTotal}</p>    
    </div>
  );
};

const mapStateToProps = (state) => { 
  const visibleExpenses = selectExpense(state.expenses, state.filters);
  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);