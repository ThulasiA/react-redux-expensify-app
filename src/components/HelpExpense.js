import React from 'react';

const HelpExpense = () => (
  <div>
    <p>Help Expense</p>
  </div>
);

export default HelpExpense;