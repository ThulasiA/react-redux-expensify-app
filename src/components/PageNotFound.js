import React from "react";
import { Link } from "react-router-dom";

const PageNotFound = () => (
  <div>
    <p>404!</p> <Link to="/"> Go Back</Link>
  </div>
);

export default PageNotFound;
