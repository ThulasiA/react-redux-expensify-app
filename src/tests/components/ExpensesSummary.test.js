import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary component', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1}  expensesTotal={2000} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary component', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={3}  expensesTotal={1234} />);
  expect(wrapper).toMatchSnapshot();
});