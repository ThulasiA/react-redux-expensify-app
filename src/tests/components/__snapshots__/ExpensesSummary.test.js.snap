// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary component 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $20.00
  </p>
</div>
`;

exports[`should render ExpensesSummary component 2`] = `
<div>
  <p>
    Viewing 
    3
     
    expenses
     totalling 
    $12.34
  </p>
</div>
`;
